### PR TITLE
Add negation to `len_zero` lint to show more explicit message.

### LINKED ISSUE #4304: Lint for `is_empty` doesn't show the required negation in the error message
```rust
fn main() {
    let v = vec![1];
    if v.len() > 0 {        
    }
}
```

Lints:

> |     if v.len() > 0 {
  |        ^^^^^^^^^^^ help: using `is_empty` is clearer and more explicit: `!v.is_empty()`
 

The problem is that if user only reads the first part of the message, they will accidentally flip the condition to `v.is_empty()`. This is a problem [reported in the user forum](https://internals.rust-lang.org/t/the-is-not-empty-method-as-more-clearly-alternative-for-is-empty/10612/19).

It'd be safer to include negation whenever `is_empty` is mentioned:

> help: using `!is_empty` is clearer and more explicit

### PR DESCRIPTION
**Pull request by chansuke**
Fixes #4304 I have updated the `len_zero` to show the required negation in case of like the below case.

```
fn main() {
    let v = vec![1];
    if v.len() > 0 {        
    }
}
```

changelog: Clarify suggestion of `len_zero` lint.

### CODE REVIEW

File: tests/ui/escape_analysis.stderr [hunk 1]
```diff
@@ -1,22 +0,0 @@
-error: local variable doesn't need to be boxed here
```
Comment (flip1995) [inline tests/ui/escape_analysis.stderr]: The issue with this test is fixed on the master branch. Please leave this file unchanged.

Comment (chansuke) [reply tests/ui/escape_analysis.stderr]: @flip1995 Thanks for the review!! Ok!I will fixe that later

Comment (chansuke) [reply tests/ui/escape_analysis.stderr]: @flip1995 I fixed it:)

**Comment by flip1995** [comment]
Oh, now you have to update the `len_zero` stderr file. :D

**Comment by chansuke** [comment]
@flip1995 Oh, my bad. I pushed updated `len_zero` stderr file:)

**Comment by flip1995** [comment]
Thanks!

@bors r+

Side note: If your contribution fixes an issue, add "Fixes/Resolves/Closes" in front of the issue number, so the issue gets closed automatically. (Also in Clippy we require a "changelog: ..."  line in the description). I added those for you, so this is just FYI and your next Clippy contribution :tada: :wink:

**Pull request merged**